internal/client/fetcher.go

```diff
@@ -149,7 +151,23 @@ func (h HTTPFetcher) ReadIssuer(ctx context.Context, hash []byte) ([]byte, error
 
 // FileFetcher knows how to fetch log artifacts from a filesystem rooted at Root.
 type FileFetcher struct {
-	Root string
+	Root                    string
+	IsEntryBundleCompressed bool
+}
+
+// decompress decompresses gzip data if needed
+func (f FileFetcher) decompress(data []byte, compressed bool) ([]byte, error) {
+	if !compressed {
+		return data, nil
+	}
+
+	reader, err := gzip.NewReader(bytes.NewReader(data))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create gzip reader: %v", err)
+	}
+	defer reader.Close()
```

Comment:
It might be a good idea to check this err - I'm not sure e.g. whether this is how checksum errors are communicated if detected.

Reply:
I checked the reader, and for consistency also made an edit to the os.ReadFile() check on L172, as it was the only place where the error was not returned formatted.